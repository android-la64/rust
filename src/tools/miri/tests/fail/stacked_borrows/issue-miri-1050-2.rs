// error-pattern: is not a valid pointer
use std::ptr::NonNull;

fn main() {
    unsafe {
        let ptr = NonNull::<i32>::dangling();
        Box::from_raw(ptr.as_ptr());
    }
}
