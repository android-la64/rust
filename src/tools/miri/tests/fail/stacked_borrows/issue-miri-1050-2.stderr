error: Undefined Behavior: 0x4 is not a valid pointer
  --> RUSTLIB/alloc/src/boxed.rs:LL:CC
   |
LL |         Box(unsafe { Unique::new_unchecked(raw) }, alloc)
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 0x4 is not a valid pointer
   |
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
           
   = note: inside `std::boxed::Box::<i32>::from_raw_in` at RUSTLIB/alloc/src/boxed.rs:LL:CC
   = note: inside `std::boxed::Box::<i32>::from_raw` at RUSTLIB/alloc/src/boxed.rs:LL:CC
note: inside `main` at $DIR/issue-miri-1050-2.rs:LL:CC
  --> $DIR/issue-miri-1050-2.rs:LL:CC
   |
LL |         Box::from_raw(ptr.as_ptr());
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to previous error

